pallets/providers/src/tests.rs

```diff
@@ -23,1040 +27,2509 @@ type DepositPerData = <Test as crate::Config>::DepositPerData;
+                            .unwrap(),
+                    );
+                    let storage_amount: StorageData<Test> = 100;
+
+                    // Get the Account Id of Alice
+                    let alice: AccountId = 0;
+
+                    // Request to sign up Alice as a Backup Storage Provider
+                    assert_ok!(StorageProviders::request_bsp_sign_up(
+                        RuntimeOrigin::signed(alice),
+                        storage_amount,
+                        multiaddresses.clone(),
+                    ));
+
+                    // Request to sign up the maximum amount of Backup Storage Providers
+                    for i in 1..<MaxBsps as Get<u32>>::get() + 1 {
+                        let account_id = i as AccountId;
+                        let account_new_balance = 1_000_000_000_000_000;
+                        assert_ok!(<Test as crate::Config>::NativeBalance::mint_into(
+                            &account_id,
+                            account_new_balance
+                        ));
+                        assert_ok!(StorageProviders::request_bsp_sign_up(
+                            RuntimeOrigin::signed(account_id),
+                            storage_amount,
+                            multiaddresses.clone(),
+                        ));
+                    }
+
+                    // Advance enough blocks for randomness to be valid
+                    run_to_block(
+                        frame_system::Pallet::<Test>::block_number()
+                            + BLOCKS_BEFORE_RANDOMNESS_VALID,
+                    );
+
+                    // Confirm the sign up of the maximum amount of Backup Storage Providers
+                    for i in 1..<MaxBsps as Get<u32>>::get() + 1 {
+                        let account_id = i as AccountId;
+                        assert_ok!(StorageProviders::confirm_sign_up(
+                            RuntimeOrigin::signed(account_id),
+                            account_id
+                        ));
+                    }
+
+                    // Try to confirm to sign up Alice as a Backup Storage Provider
+                    assert_noop!(
+                        StorageProviders::confirm_sign_up(RuntimeOrigin::signed(alice), alice,),
+                        Error::<Test>::MaxBspsReached
+                    );
+                });
+            }
+        }
 
-            // Try to sign up Alice as a Main Storage Provider with less than the minimum capacity
-            assert_noop!(
-                StorageProviders::msp_sign_up(
-                    RuntimeOrigin::signed(alice),
-                    storage_amount,
-                    multiaddresses.clone(),
-                    value_prop.clone()
-                ),
-                Error::<Test>::StorageTooLow
-            );
+        /// This module holds the failure cases for functions that test both Main Storage Providers and Backup Storage Providers
+        mod msp_and_bsp {
```

Comment:
After adding the check for if an AccountId already has a registration request, add a failure check for it, with an account trying to first sign up as BSP, and then MSP, or viceversa.